modyn/selector/internal/selector_strategies/new_data_strategy.py

```diff
@@ -131,7 +143,31 @@ def _get_current_trigger_data(self) -> Iterable[list[int]]:
                 .execution_options(yield_per=self._maximum_keys_in_memory)
                 .filter(
                     SelectorStateMetadata.pipeline_id == self._pipeline_id,
-                    SelectorStateMetadata.seen_in_trigger_id == self._next_trigger_id,
+                    self._next_trigger_id == SelectorStateMetadata.seen_in_trigger_id,
+                )
+                .order_by(asc(SelectorStateMetadata.timestamp))
+            )
+
+            for chunk in database.session.execute(stmt).partitions():
+                if len(chunk) > 0:
+                    yield [res[0] for res in chunk]
+                else:
+                    yield []
+
+    def _get_tail_triggers_data(self) -> Iterable[list[int]]:
+        """Returns all sample for current trigger
+
+        Returns:
+            list[int]: Keys of used samples
+        """
+        with MetadataDatabaseConnection(self._modyn_config) as database:
+            stmt = (
+                select(SelectorStateMetadata.sample_key)
+                # Enables batching of results in chunks. See https://docs.sqlalchemy.org/en/20/orm/queryguide/api.html#orm-queryguide-yield-per
+                .execution_options(yield_per=self._maximum_keys_in_memory)
+                .filter(
+                    SelectorStateMetadata.pipeline_id == self._pipeline_id,
+                    self._next_trigger_id - SelectorStateMetadata.seen_in_trigger_id <= self.tail_triggers,
```

Comment:
Could you rephrase this as conditions on SelectorStateMetadata? Like SelectorStateMetadata. seen_in_trigger_id >= - self.tail_triggers - self._next_trigger_id or something that is maybe a bit more readable? It's a bit more easy to read to condition on SelectorStateMetadata I think.